platform/ios/src/MGLMapView.mm

```diff
@@ -1772,7 +1772,13 @@ - (void)handleSingleTapGesture:(UITapGestureRecognizer *)singleTap
     }
     else if (self.selectedAnnotation)
     {
-        [self deselectAnnotation:self.selectedAnnotation animated:YES];
+        BOOL deselect = YES;
+        if ([self.selectedAnnotation respondsToSelector:@selector(dismissesAutomatically)]) {
+            deselect = [self.selectedAnnotation performSelector:@selector(dismissesAutomatically)];
```

Comment:
This is a backwards-incompatible change, given the following documented behavior of `dismissesAutomatically`:

https://github.com/mapbox/mapbox-gl-native/blob/2b84c2df27c2a1c01699a17038644fb4c111a27c/platform/ios/src/MGLCalloutView.h#L94-L95

I think we’d need to add a separate property if we want to take this approach.

Reply:
I think we should rename this property `deselectsAutomatically`.

To @1ec5's point - we should also look at the interaction with the callout's `dismissesAutomatically` property - which currently controls both the callout's removal AND the annotation's deselection. This might just be a case of improving the documentation - though having a quick look we will need to think about what happens in `-cameraWillChangeAnimated:`